src/domain/fiat/index.types.d.ts

```diff
@@ -1,2 +1,47 @@
 type UsdCents = number & { readonly brand: unique symbol }
 type DisplayCurrencyBaseAmount = number & { readonly brand: unique symbol }
+
+type CurrencyBaseAmount = Satoshis | UsdCents
+
+interface DisplayCurrencyConversionRate {
+  fromSats: (amount: Satoshis) => DisplayCurrencyBaseAmount
+  fromCents: (amount: UsdCents) => DisplayCurrencyBaseAmount
+}
+
+interface DealerFns {
+  buyUsdImmediate: (amount: Satoshis) => Promise<UsdCents>
+  buyUsdImmediateFromCents: (amount: UsdCents) => Promise<Satoshis>
+  sellUsdImmediate: (amount: UsdCents) => Promise<Satoshis>
+  sellUsdImmediateFromSats: (amount: Satoshis) => Promise<UsdCents>
+  getBuyUsdQuoteFromCents: (amount: UsdCents) => Promise<Satoshis>
+}
+
+interface AmountFromSatoshis {
+  sats: Satoshis
+}
+
+interface AmountFromCents {
+  cents: UsdCents
+}
+
+type XOR<T1, T2> =
+  | (T1 & { [k in Exclude<keyof T2, keyof T1>]?: never })
+  | (T2 & { [k in Exclude<keyof T1, keyof T2>]?: never })
+
+type CentsXORSats = XOR<AmountFromSatoshis, AmountFromCents>
+
+type OrderType = "immediate" | "quote"
+
+type GetAmountsSendOrReceiveArgs = {
+  walletCurrency: WalletCurrency
+  order: OrderType
+} & CentsXORSats
```

Comment:
You are unifying 2 different call signatures into 1 quasi overloaded function. Cents XOR sats. In cases where:
- the calling code knows which variant is being called (it does)
- the function must use conditionals to separate code paths depending on which is being called (it does)
- there is no or very little shared code between the 2 code paths (there is none)
it is generally preferable to segregate the 2 signatures into separate functions.

In this case it would:
- simplify the type
- remove a conditional
- probably result in less code overall